sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java

```diff
@@ -187,11 +188,12 @@ public BeamSqlRowComparator(List<Integer> fieldsIndices,
       this.nullsFirst = nullsFirst;
     }
 
-    @Override public int compare(BeamSqlRow row1, BeamSqlRow row2) {
+    @Override public int compare(BeamRecord row1, BeamRecord row2) {
       for (int i = 0; i < fieldsIndices.size(); i++) {
         int fieldIndex = fieldsIndices.get(i);
         int fieldRet = 0;
-        SqlTypeName fieldType = CalciteUtils.getFieldType(row1.getDataType(), fieldIndex);
+        SqlTypeName fieldType = CalciteUtils.getFieldType(
```

Comment:
Based on the implementation below, couldn't you just call row1.get(fieldIndex).compareTo(row2.get(fieldIndex)) iff they're instances of Comparable, and raise UnsupportedOperationException otherwise?